mujoco_warp/_src/io.py

```diff
@@ -461,515 +278,314 @@ def not_implemented(objtype, objid, geomtype):
+    values = mjm.plugin_attr[start:end]
+    attr_values = []
+    current = []
+    for v in values:
+      if v == 0:
+        if current:
+          s = "".join(chr(int(x)) for x in current)
+          attr_values.append(float(s))
+          current = []
+      else:
+        current.append(v)
+    # Pad with zeros if less than 3
+    attr_values += [0.0] * (3 - len(attr_values))
+    m.plugin_attr.append(attr_values[:3])
+
+  # equality constraint addresses
+  m.eq_connect_adr = np.nonzero(mjm.eq_type == types.EqType.CONNECT)[0]
+  m.eq_wld_adr = np.nonzero(mjm.eq_type == types.EqType.WELD)[0]
+  m.eq_jnt_adr = np.nonzero(mjm.eq_type == types.EqType.JOINT)[0]
+  m.eq_ten_adr = np.nonzero(mjm.eq_type == types.EqType.TENDON)[0]
+
+  # fixed tendon
+  m.tendon_jnt_adr, m.wrap_jnt_adr = [], []
+  for i in range(mjm.ntendon):
+    adr = mjm.tendon_adr[i]
+    if mjm.wrap_type[adr] == mujoco.mjtWrap.mjWRAP_JOINT:
+      tendon_num = mjm.tendon_num[i]
+      for j in range(tendon_num):
+        m.tendon_jnt_adr.append(i)
+        m.wrap_jnt_adr.append(adr + j)
+
+  # spatial tendon
+  m.tendon_site_pair_adr, m.tendon_geom_adr = [], []
+  m.ten_wrapadr_site, m.ten_wrapnum_site = [0], []
+  for i, tendon_num in enumerate(mjm.tendon_num):
+    adr = mjm.tendon_adr[i]
+    # sites
+    if (mjm.wrap_type[adr : adr + tendon_num] == mujoco.mjtWrap.mjWRAP_SITE).all():
+      if i < mjm.ntendon:
+        m.ten_wrapadr_site.append(m.ten_wrapadr_site[-1] + tendon_num)
+      m.ten_wrapnum_site.append(tendon_num)
+    else:
+      if i < mjm.ntendon:
+        m.ten_wrapadr_site.append(m.ten_wrapadr_site[-1])
+      m.ten_wrapnum_site.append(0)
+
+    # geoms
+    for j in range(tendon_num):
+      wrap_type = mjm.wrap_type[adr + j]
+      if j < tendon_num - 1:
+        next_wrap_type = mjm.wrap_type[adr + j + 1]
+        if wrap_type == mujoco.mjtWrap.mjWRAP_SITE and next_wrap_type == mujoco.mjtWrap.mjWRAP_SITE:
+          m.tendon_site_pair_adr.append(i)
+      if wrap_type == mujoco.mjtWrap.mjWRAP_SPHERE or wrap_type == mujoco.mjtWrap.mjWRAP_CYLINDER:
+        m.tendon_geom_adr.append(i)
+
+  m.tendon_limited_adr = np.nonzero(mjm.tendon_limited)[0]
+  m.wrap_site_adr = np.nonzero(mjm.wrap_type == mujoco.mjtWrap.mjWRAP_SITE)[0]
+  m.wrap_site_pair_adr = np.setdiff1d(m.wrap_site_adr[np.nonzero(np.diff(m.wrap_site_adr) == 1)[0]], mjm.tendon_adr[1:] - 1)
+  m.wrap_geom_adr = np.nonzero(np.isin(mjm.wrap_type, [mujoco.mjtWrap.mjWRAP_SPHERE, mujoco.mjtWrap.mjWRAP_CYLINDER]))[0]
+
+  # pulley scaling
+  m.wrap_pulley_scale = np.ones(mjm.nwrap, dtype=float)
+  pulley_adr = np.nonzero(mjm.wrap_type == mujoco.mjtWrap.mjWRAP_PULLEY)[0]
+  for tadr, tnum in zip(mjm.tendon_adr, mjm.tendon_num):
+    for padr in pulley_adr:
+      if tadr <= padr < tadr + tnum:
+        m.wrap_pulley_scale[padr : tadr + tnum] = 1.0 / mjm.wrap_prm[padr]
+
+  # actuator_moment tiles are grouped by dof size and number of actuators
```

Comment:
the fields `actuator_moment_tiles_{nu,nv}` are unused. we can remove this section and these fields from types.py?